Jenkinsfile

```diff
@@ -19,15 +17,16 @@ def pkgs = [
     [target: "ubuntu-jammy",             image: "ubuntu:jammy",                           arches: ["amd64", "aarch64", "armhf"]], // Ubuntu 22.04 LTS (End of support: April, 2027. EOL: April, 2032)
 ]
 
-def genBuildStep(LinkedHashMap pkg, String arch) {
+def statics = [
+    [os: "linux",       arches: ["amd64", "armv6", "armv7", "aarch64"]],
+    [os: "darwin",      arches: ["amd64", "aarch64"]],
+    [os: "windows",     arches: ["amd64"]],
+]
+
+def genPkgStep(LinkedHashMap pkg, String arch) {
     def nodeLabel = "linux&&${arch}"
-    def platform = ""
     def branch = env.CHANGE_TARGET ?: env.BRANCH_NAME
-
     if (arch == 'armhf') {
-        // Running armhf builds on EC2 requires --platform parameter
-        // Otherwise it accidentally pulls armel images which then breaks the verify step
-        platform = "--platform=linux/${arch}"
```

Comment:
this no longer needed? was slightly wondering if we must do the reverse and always explicitly specify platform 🤔

Reply:
we could be we can't with moby atm. only cli, buildx supports it without effort. hence https://github.com/docker/docker-ce-packaging/pull/665#issuecomment-1081864970. if it's fixed on moby we can remove the `docker_engine_cross` logic in https://github.com/docker/docker-ce-packaging/pull/665/files#diff-6143272ec6e2f5ecf2b4cb74959addf4eff0bd7839427318937c70c983740461R34-R79